src/core/cgroup.c

```diff
@@ -1164,7 +1164,7 @@ static void cgroup_apply_restrict_network_interfaces(Unit *u) {
 }
 
 static int cgroup_apply_devices(Unit *u) {
-        _cleanup_(bpf_program_unrefp) BPFProgram *prog = NULL;
+        BPFProgram *prog = NULL;
```

Comment:
This will cause memory leak on the error path in between `bpf_devices_cgroup_init` (when `prog` is initialized) and `bpf_devices_apply_policy` (when `prog` is assigned to `bpf_device_control_installed`) calls.
Luckily, there is no `return` in between... sill this change has this potential danger.

What about leaving `_cleanup_`:
```
_cleanup_(bpf_program_freep) BPFProgram *prog = NULL;`
```

Modifying `bpf_devices_apply_policy` signature, so it accepts pointer-to-pointer:

```
bpf_devices_apply_policy(
        BPFProgram **prog,
         /*...*/
``` 

`TAKE_PTR` when assigning `prog` to `bpf_device_control_installed`:

```
 *prog_installed = TAKE_PTR(*prog);
```